common/lib/client/push.js

```diff
@@ -0,0 +1,134 @@
+var Push = (function() {
+	var msgpack = (typeof require !== 'function') ? Ably.msgpack : require('msgpack-js');
+
+	function Push(rest) {
+		this.rest = rest;
+		this.admin = new Admin(rest);
+	}
+
+	Push.prototype.publish = function(recipient, payload, callback) {
+		var rest = this.rest;
+		var format = rest.options.useBinaryProtocol ? 'msgpack' : 'json',
+		    requestBody = Utils.mixin({recipient: recipient}, payload),
+		    headers = Utils.copy(Utils.defaultPostHeaders(format));
```

Comment:
`Utils.defaultPostHeaders` creates and returns a new object, so I don't think you need the `Utils.copy`